### DISCUSSION TITLE
automatic indexing of originals not working/running

### DISCUSSION THREAD

**Discussion by hillbicks** [photoprism/photoprism#1952] [Q&A] [answered]
Hey all,

first of all, what a fantastic piece of software. I'm really impressed with photoprosm and so is my wife. It is so good, that I finally took the time to organize my photo collection. But anyway.....

I'm running the docker version 220107-f5b7ef83-Linux-x86_64. 

Photos are solely taken from originals directory, no import of files is done so to speak. Initial indexing of originals went fine, everything got imported. But the automatic indexing of the originals doesn't seem to be working. Once a new picture arrives in /photoprism/originals, it will only show up in photoprism after manually pressing the "start indexing" button. 

When start indexing is pressed, I can see 

` 2022-01-17 12:12:21 INFO indexing originals...`

If I don't start it manually, the only log entries I see are: 
```
2022-01-17 12:19:43 INFO index: updating covers
2022-01-17 12:19:43 INFO index: updating counts
```

and therefore new images are not appearing. My understanding is, that the automatic index is run according to the interval set by PHOTOPRISM_AUTO_INDEX (in my case 300). Is there anything else I have to set in order for it to run automatically?

docker envs

```
PHOTOPRISM_ADMIN_PASSWORD | XXXX
-- | --
PHOTOPRISM_ASSETS_PATH | /photoprism/assets
PHOTOPRISM_AUTO_IMPORT | 300
PHOTOPRISM_AUTO_INDEX | 300
PHOTOPRISM_BACKUP_PATH | /var/lib/photoprism
PHOTOPRISM_DARKTABLE_PRESETS | false
PHOTOPRISM_DATABASE_DRIVER | sqlite
PHOTOPRISM_DATABASE_NAME | photoprism
PHOTOPRISM_DATABASE_PASSWORD |  
PHOTOPRISM_DATABASE_SERVER |  
PHOTOPRISM_DATABASE_USER | photoprism
PHOTOPRISM_DEBUG | false
PHOTOPRISM_DETECT_NSFW | false
PHOTOPRISM_DISABLE_BACKUPS | false
PHOTOPRISM_DISABLE_CHOWN | false
PHOTOPRISM_DISABLE_CLASSIFICATION | false
PHOTOPRISM_DISABLE_EXIFTOOL | false
PHOTOPRISM_DISABLE_FACES | false
PHOTOPRISM_DISABLE_PLACES | false
PHOTOPRISM_DISABLE_SETTINGS | false
PHOTOPRISM_DISABLE_TENSORFLOW | false
PHOTOPRISM_DISABLE_WEBDAV | false
PHOTOPRISM_EXPERIMENTAL | false
PHOTOPRISM_HTTP_HOST | 0.0.0.0
PHOTOPRISM_HTTP_PORT | 2342
PHOTOPRISM_IMPORT_PATH | /photoprism/import
PHOTOPRISM_JPEG_QUALITY | 92
PHOTOPRISM_JPEG_SIZE | 7680
PHOTOPRISM_LOG_FILENAME | /photoprism/photoprism.log
PHOTOPRISM_ORIGINALS_LIMIT | 10000
PHOTOPRISM_ORIGINALS_PATH | /photoprism/originals
PHOTOPRISM_PID_FILENAME | /photoprism/photoprism.pid
PHOTOPRISM_PUBLIC | false
PHOTOPRISM_READONLY | false
PHOTOPRISM_SITE_AUTHOR |  
PHOTOPRISM_SITE_CAPTION | Browse Your Life
PHOTOPRISM_SITE_DESCRIPTION |  
PHOTOPRISM_SITE_TITLE | PhotoPrism
PHOTOPRISM_SITE_URL | http://localhost:2342/
PHOTOPRISM_STORAGE_PATH | /photoprism/storage
PHOTOPRISM_THUMB_FILTER | lanczos
PHOTOPRISM_THUMB_SIZE | 2048
PHOTOPRISM_THUMB_SIZE_UNCACHED | 7680
PHOTOPRISM_THUMB_UNCACHED | false
PHOTOPRISM_UPLOAD_NSFW | true
PHOTOPRISM_WAKEUP_INTERVAL | 900
PHOTOPRISM_WORKERS | 0
TF_CPP_MIN_LOG_LEVEL | 2
TMPDIR | /tmp
```
Thanks again!

**Comment by graciousgrey** [comment, marked as answer]
Thank you very much for your feedback!

>  My understanding is, that the automatic index is run according to the interval set by PHOTOPRISM_AUTO_INDEX (in my case 300). Is there anything else I have to set in order for it to run automatically?

Automatic indexing is only triggered when files have been added to originals using WebDAV (https://docs.photoprism.app/user-guide/library/originals/#automatic-indexing). If you don't use WebDAV to add files to your originals directory you need to use the scheduler to run cron jobs, see: https://docs.photoprism.app/getting-started/docker-compose/#command-line-interface & https://dl.photoprism.app/docker/scheduler/

**Reply by hillbicks** [reply]
Ah, that makes sense. Sorry, didn't see the docker specification in the documentation. For anybody else looking at this, here is how I solved it (systemd service and a timer)

**photoprism_index.service**
```
[Unit]
Description=Photoprism run index
After=docker.service
Requires=docker.service

[Service]
Type=simple
ExecStart=/usr/bin/docker exec  photoprism photoprism index --cleanup
User=root
KillMode=process
Restart=always

[Install]
WantedBy=multi-user.target
```
**photoprism_index.timer**

```
[Unit]
Description=Run photoprism indexer

[Timer]
OnCalendar=*:0/5
Persistent=true

[Install]
WantedBy=timers.target
```
Create those two files and then enable and start the timer. It will start the service every 5 minutes. Don't enable the service file, it will get called by the enabled timer
`sudo systemctl enable photoprism_index.timer`
`sudo systemctl start photoprism_index.timer`

You can check the next run with 
`sudo systemctl list-timers`

Tested on my system and it works.

**Reply by hillbicks** [reply]
Just to add to that, I created two separate service files and corresponding timers to run the face index and convert once a day during the night.

**Reply by thierrybla** [reply]
Hi, I want to use this but how do you use this with docker @hillbicks ? I don't want to use the official scheduler because it needs me to add another container.

**Reply by hillbicks** [reply]
Hi @thierrybla 

not sure what your question is exactly, but I'll try to answer. 

On the host that is running the docker photoprism command, add additional service files for each command you want to run inside the docker container. Then add the correspondent timer file that starts the service. If you're not familiar with systemd timers, here is a good place to start. 

https://wiki.archlinux.org/title/Systemd/Timers

You could also use cron of course, whatever fits your need the best.

**Reply by Ramblurr** [reply]
The provided systemd unit files will not work as you expect. `Type=simple` and `Restart=always` will cause the jobs to run continuously. When they exit, systemd will start it again.

Instead you want a `oneshot` job with no restart policy:


Example (change for your own usage)
```
[Unit]
Description=Photoprism run video converter
After=pod-pp.service
Requires=pod-pp.service

[Service]
Type=oneshot   # <-- this is the important bit
ExecStart=/usr/bin/podman exec pp-photoprism photoprism convert
User=root
KillMode=process

[Install]
WantedBy=multi-user.target
```
